### DISCUSSION TITLE
optionalDependences in registry not the same as package.json

### DISCUSSION THREAD

**Discussion by kwasimensah** [npm/feedback#1057] [Registry]
https://registry.npmjs.org/esbuild/0.19.10 lists all the packages in the `optionalDependencies` field  in the `dependencies` field which is different than the package.json in the tarball (the same as https://github.com/evanw/esbuild/blob/main/npm/esbuild/package.json). Is there a reason for the discrepancy?

I'm working on a build tool and trying to reason about the optional dependencies of a package using metadata so I don't have to download full package sand then reject them

**Comment by ljharb** [comment]
You always need to download the full package; see https://blog.vlt.sh/blog/the-massive-hole-in-the-npm-ecosystem

**Comment by kwasimensah** [comment]
Thanks. This is also a bit scary because I was using the list of versions
in the registry to resolve version specifiers. I think I’m still fine for
that use case but I’ll switch to full downloads in the future and point
people to this issue for the reason.

Thanks

**Reply by ljharb** [reply]
You definitely can rely on the version metadata, it's just the packument contents that come from the publisher that you can't trust.
